include/stl2/detail/iterator/operations.hpp

```diff
@@ -82,10 +82,10 @@ STL2_OPEN_NAMESPACE {
 			} else {
 				if constexpr (BidirectionalIterator<I> && Same<I, S>) {
 					if (0 > n) {
-						do {
+						while (n != 0 && i != bound) {
```

Comment:
Inconsistent the ordering of `n != 0` is. Un-Yoda the other comparisons in this function we should as well.